pkg/oc/cli/secrets/options.go

```diff
@@ -98,11 +99,26 @@ func (o SecretOptions) GetServiceAccount() (*kapi.ServiceAccount, error) {
 func (o SecretOptions) GetSecretNames(secrets []*kapi.Secret) sets.String {
 	names := sets.String{}
 	for _, secret := range secrets {
-		names.Insert(secret.Name)
+		names.Insert(parseSecretName(secret.Name))
 	}
 	return names
 }
 
+// parseSecretName receives a resource name as either
+// <resource type> / <name> or <name> and returns only the resource <name>.
+func parseSecretName(name string) string {
+	segs := strings.Split(name, "/")
+	if len(segs) < 2 {
+		return name
+	}
+
+	if segs[0] == "secret" || segs[0] == "secrets" {
```

Comment:
If you have a slash and this is not true, there is an error, right?

Reply:
Right, the name will be returned as-is (e.g. `pods/my-secret`), and the user will end up seeing 
```
secret "pod/my-secret" not found
error: No valid secrets found or secrets not linked to service account
```

Reply:
OK. I meant more forcefully -- the error _really_ is that a secret name cannot start with `pod/` not that `pod/my-secret` was not found

Reply:
Sounds good, added a check for this in the `Validate` method